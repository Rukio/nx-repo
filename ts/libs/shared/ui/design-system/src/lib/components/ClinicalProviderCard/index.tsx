import { useCallback } from 'react';
import { Box, Button, Card, Typography } from '../..';
import { makeSxStyles } from '../../utils/makeSxStyles';
import { formatPhoneNumber } from '../../utils/formatters';
import CLINICAL_PROVIDER_CARD_TEST_IDS from './testIds';

export enum ClinicalProviderCardLayoutDirection {
  horizontal = 'HORIZONTAL',
  vertical = 'VERTICAL',
}
export interface ClinicalProviderDetails {
  id: number;
  name: string;
  address: string;
  city: string;
  state: string;
  zipCode: string;
  fax?: string;
  phone: string;
}

export interface ClinicalProviderCardProps {
  clinicalProviderDetails: ClinicalProviderDetails;
  isFaxVisible: boolean;
  onSelect?: (selectedId: number) => void;
  testIdPrefix: string;
  layoutDirection?: ClinicalProviderCardLayoutDirection;
}

const makeStyles = (layoutDirection: ClinicalProviderCardLayoutDirection) =>
  makeSxStyles({
    capitalize: {
      textTransform: 'capitalize',
    },
    card: {
      width: '100%',
      minHeight: '132px',
    },
    cardContent: {
      padding: 2,
      paddingBottom: 2,
      display: 'flex',
      flexDirection:
        layoutDirection === ClinicalProviderCardLayoutDirection.horizontal
          ? 'row'
          : 'column',
      alignItems:
        layoutDirection === ClinicalProviderCardLayoutDirection.horizontal
          ? 'center'
          : 'flex-start',
      justifyContent: 'space-between',
    },
    addressContainer: {
      marginBottom: 1,
    },
    label: (theme) => ({
      color: theme.palette.text.secondary,
      marginRight: 1,
    }),
    selectButton: {
      width: '100%',
      marginTop: 1,
    },
  });

const ClinicalProviderCard: React.FC<ClinicalProviderCardProps> = ({
  clinicalProviderDetails,
  isFaxVisible,
  onSelect,
  testIdPrefix,
  layoutDirection = ClinicalProviderCardLayoutDirection.horizontal,
}) => {
  const { id, name, address, city, state, zipCode, phone, fax } =
    clinicalProviderDetails;
  const styles = makeStyles(layoutDirection);

  const handleSelect = useCallback(() => onSelect?.(id), [onSelect, id]);

  const formatPhone = (telephoneNumber: string) =>
    telephoneNumber ? formatPhoneNumber(telephoneNumber) : 'None';

  const getTestIdFor = (testId: string): string => `${testIdPrefix}-${testId}`;

  return (
    <Card
      variant="outlined"
      sx={styles.card}
      data-testid={`${getTestIdFor(
        CLINICAL_PROVIDER_CARD_TEST_IDS.CLINICAL_PROVIDER_CARD
      )}-${clinicalProviderDetails.id}`}
    >
      <Box sx={styles.cardContent}>
        <Box>
          <Typography
            data-testid={getTestIdFor(
              CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_NAME
            )}
            variant="h6"
          >
            {name.toUpperCase()}
          </Typography>
          <Box sx={styles.addressContainer}>
            <Typography
              variant="body2"
              data-testid={getTestIdFor(
                CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_ADDRESS_LINE_1
              )}
              sx={styles.capitalize}
            >
              {address.toLowerCase()}
            </Typography>
            <Typography
              variant="body2"
              data-testid={getTestIdFor(
                CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_CITY_STATE
              )}
              sx={styles.capitalize}
            >
              {city.toLowerCase()}, {state.toUpperCase()} {zipCode}
            </Typography>
          </Box>
          <Box
            data-testid={getTestIdFor(
              CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_PHONE_CONTAINER
            )}
          >
            <Typography variant="body2" display="inline" sx={styles.label}>
              Phone
            </Typography>
            <Typography
              variant="body2"
              display="inline"
              data-testid={getTestIdFor(
                CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_PHONE
              )}
            >
              {formatPhone(phone)}
            </Typography>
          </Box>
          {isFaxVisible && (
            <Box
              data-testid={getTestIdFor(
                CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX_CONTAINER
              )}
            >
              <>
                <Typography variant="body2" display="inline" sx={styles.label}>
                  Fax
                </Typography>
                <Typography
                  variant="body2"
                  display="inline"
                  data-testid={getTestIdFor(
                    CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX
                  )}
                >
                  {formatPhone(fax || '')}
                </Typography>
              </>
            </Box>
          )}
        </Box>
        {onSelect && (
          <Button
            size="large"
            variant="outlined"
            color="primary"
            data-testid={getTestIdFor(
              CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_SELECT_BUTTON
            )}
            onClick={handleSelect}
            sx={
              layoutDirection === ClinicalProviderCardLayoutDirection.vertical
                ? styles.selectButton
                : null
            }
          >
            Select
          </Button>
        )}
      </Box>
    </Card>
  );
};
export default ClinicalProviderCard;
