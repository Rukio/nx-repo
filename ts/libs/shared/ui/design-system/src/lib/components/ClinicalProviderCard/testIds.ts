const CLINICAL_PROVIDER_CARD_TEST_IDS = {
  // test id strings will be prefixed by the testIdPrefix prop (ex. xyz-clinical-provider-card)
  CLINICAL_PROVIDER_CARD: 'clinical-provider-card',
  PROVIDER_NAME: 'provider-name',
  PROVIDER_ADDRESS_LINE_1: 'provider-address-line1',
  PROVIDER_CITY_STATE: 'provider-city-state',
  PROVIDER_PHONE_CONTAINER: 'provider-phone-container',
  PROVIDER_PHONE: 'provider-phone',
  PROVIDER_FAX_CONTAINER: 'provider-fax-container',
  PROVIDER_FAX: 'provider-fax',
  PROVIDER_SELECT_BUTTON: 'provider-select-button',
};

export default CLINICAL_PROVIDER_CARD_TEST_IDS;
