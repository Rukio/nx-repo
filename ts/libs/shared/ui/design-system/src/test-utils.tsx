import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
