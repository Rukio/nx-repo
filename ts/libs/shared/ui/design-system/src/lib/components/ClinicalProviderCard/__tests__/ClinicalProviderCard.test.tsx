import { render, screen } from '../../../../test-utils';
import { Box } from '../../..';
import CLINICAL_PROVIDER_CARD_TEST_IDS from '../testIds';

import ClinicalProviderCard, {
  ClinicalProviderDetails,
  ClinicalProviderCardProps,
  ClinicalProviderCardLayoutDirection,
} from '../index';

const mockClinicalProvider: ClinicalProviderDetails = {
  id: 42,
  name: 'theodor GEISEL',
  address: '7301 ENCELIA drive',
  city: 'la JOlla',
  state: 'cA',
  zipCode: '92037',
  phone: '999 555 1212',
  fax: '999 555 1000',
};

interface SetupInterface {
  clinicalProviderCardProps: ClinicalProviderCardProps;
  containerWidthPixels: number;
}

const setup = ({
  clinicalProviderCardProps,
  containerWidthPixels,
}: SetupInterface) => {
  const {
    clinicalProviderDetails,
    isFaxVisible,
    onSelect,
    testIdPrefix,
    layoutDirection,
  } = clinicalProviderCardProps;

  return render(
    <Box sx={{ width: `${containerWidthPixels}px` }}>
      <ClinicalProviderCard
        clinicalProviderDetails={clinicalProviderDetails}
        isFaxVisible={isFaxVisible}
        onSelect={onSelect}
        testIdPrefix={testIdPrefix}
        layoutDirection={layoutDirection}
      />
    </Box>
  );
};

const getTestIdFor = (testId: string): string => `test-${testId}`;

describe('ClinicalProviderCard', () => {
  it('should render correctly horizontally with fax and test ids', () => {
    const mockOnSelect = jest.fn();
    const { asFragment } = setup({
      clinicalProviderCardProps: {
        clinicalProviderDetails: mockClinicalProvider,
        isFaxVisible: true,
        onSelect: mockOnSelect,
        testIdPrefix: 'test',
      },
      containerWidthPixels: 500,
    });
    expect(
      screen.getByTestId(
        `${getTestIdFor(
          CLINICAL_PROVIDER_CARD_TEST_IDS.CLINICAL_PROVIDER_CARD
        )}-${mockClinicalProvider.id}`
      )
    ).toBeVisible();
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_NAME)
      )
    ).toHaveTextContent(mockClinicalProvider.name.toUpperCase());
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_ADDRESS_LINE_1)
      )
    ).toHaveTextContent(mockClinicalProvider.address.toLowerCase());
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_CITY_STATE)
      )
    ).toHaveTextContent(
      `${mockClinicalProvider.city.toLowerCase()}, ${mockClinicalProvider.state.toUpperCase()} ${
        mockClinicalProvider.zipCode
      }`
    );
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_PHONE)
      )
    ).toHaveTextContent('(999) 555-1212');
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX_CONTAINER)
      )
    ).toBeVisible();
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX)
      )
    ).toHaveTextContent('(999) 555-1000');
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_SELECT_BUTTON)
      )
    ).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly vertically', () => {
    const mockOnSelect = jest.fn();
    const { asFragment } = setup({
      clinicalProviderCardProps: {
        clinicalProviderDetails: mockClinicalProvider,
        isFaxVisible: true,
        onSelect: mockOnSelect,
        testIdPrefix: 'test',
        layoutDirection: ClinicalProviderCardLayoutDirection.vertical,
      },
      containerWidthPixels: 300,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should not show fax is display is disabled', () => {
    const mockOnSelect = jest.fn();
    setup({
      clinicalProviderCardProps: {
        clinicalProviderDetails: mockClinicalProvider,
        isFaxVisible: false,
        onSelect: mockOnSelect,
        testIdPrefix: 'test',
      },
      containerWidthPixels: 500,
    });
    expect(
      screen.queryByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX_CONTAINER)
      )
    ).not.toBeInTheDocument();
  });

  it('should show none for fax if no fax value', () => {
    const mockOnSelect = jest.fn();
    setup({
      clinicalProviderCardProps: {
        clinicalProviderDetails: { ...mockClinicalProvider, fax: '' },
        isFaxVisible: true,
        onSelect: mockOnSelect,
        testIdPrefix: 'test',
      },
      containerWidthPixels: 500,
    });
    expect(
      screen.getByTestId(
        getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_FAX_CONTAINER)
      )
    ).toBeVisible();
    expect(screen.getByTestId('test-provider-fax')).toHaveTextContent('None');
  });

  it('should be selectable', async () => {
    const mockOnSelect = jest.fn();
    const { user } = setup({
      clinicalProviderCardProps: {
        clinicalProviderDetails: mockClinicalProvider,
        isFaxVisible: true,
        onSelect: mockOnSelect,
        testIdPrefix: 'test',
      },
      containerWidthPixels: 500,
    });
    const selectButton = screen.getByTestId(
      getTestIdFor(CLINICAL_PROVIDER_CARD_TEST_IDS.PROVIDER_SELECT_BUTTON)
    );
    expect(selectButton).toHaveTextContent('Select');
    await user.click(selectButton);
    expect(mockOnSelect).toHaveBeenCalledWith(mockClinicalProvider.id);
  });
});
