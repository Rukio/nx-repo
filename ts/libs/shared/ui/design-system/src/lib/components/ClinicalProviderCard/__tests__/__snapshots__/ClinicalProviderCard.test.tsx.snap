// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicalProviderCard should render correctly horizontally with fax and test ids 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 500px;
}

.emotion-1 {
  background-color: #FEFEFE;
  color: #1E2930;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 6px;
  border: 1px solid #EFF4F7;
  overflow: hidden;
  width: 100%;
  min-height: 132px;
}

.emotion-2 {
  padding: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 {
  margin: 0;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.375;
  letter-spacing: 0.01rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
}

.emotion-5 {
  margin-bottom: 8px;
}

.emotion-6 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  text-transform: capitalize;
}

.emotion-9 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  display: inline;
  color: #5E696F;
  margin-right: 8px;
}

.emotion-10 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  display: inline;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-weight: 600;
  font-size: 0.9375rem;
  line-height: 1.714;
  text-transform: none;
  letter-spacing: 0.025rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  min-width: 64px;
  padding: 7px 21px;
  border-radius: 6px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(0, 116, 184, 0.5);
  color: #0074B8;
  box-shadow: none;
}

.emotion-14::-moz-focus-inner {
  border-style: none;
}

.emotion-14.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-14 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 116, 184, 0.04);
  border: 1px solid #0074B8;
}

@media (hover: none) {
  .emotion-14:hover {
    background-color: transparent;
  }
}

.emotion-14.Mui-disabled {
  color: rgba(17, 27, 34, 0.26);
  border: 1px solid rgba(17, 27, 34, 0.12);
}

.emotion-14:hover {
  box-shadow: none;
}

.emotion-14.Mui-focusVisible {
  box-shadow: none;
}

.emotion-14:active {
  box-shadow: none;
}

.emotion-14.Mui-disabled {
  box-shadow: none;
}

.emotion-15 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root emotion-1"
      data-testid="test-clinical-provider-card-42"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <div
          class="MuiBox-root emotion-3"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 emotion-4"
            data-testid="test-provider-name"
          >
            THEODOR GEISEL
          </h6>
          <div
            class="MuiBox-root emotion-5"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-6"
              data-testid="test-provider-address-line1"
            >
              7301 encelia drive
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-6"
              data-testid="test-provider-city-state"
            >
              la jolla, CA 92037
            </p>
          </div>
          <div
            class="MuiBox-root emotion-3"
            data-testid="test-provider-phone-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-9"
            >
              Phone
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-10"
              data-testid="test-provider-phone"
            >
              (999) 555-1212
            </p>
          </div>
          <div
            class="MuiBox-root emotion-3"
            data-testid="test-provider-fax-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-9"
            >
              Fax
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-10"
              data-testid="test-provider-fax"
            >
              (999) 555-1000
            </p>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-disableElevation emotion-14"
          data-testid="test-provider-select-button"
          tabindex="0"
          type="button"
        >
          Select
          <span
            class="MuiTouchRipple-root emotion-15"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ClinicalProviderCard should render correctly vertically 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 300px;
}

.emotion-1 {
  background-color: #FEFEFE;
  color: #1E2930;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 6px;
  border: 1px solid #EFF4F7;
  overflow: hidden;
  width: 100%;
  min-height: 132px;
}

.emotion-2 {
  padding: 16px;
  padding-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-4 {
  margin: 0;
  font-weight: 600;
  font-size: 1rem;
  line-height: 1.375;
  letter-spacing: 0.01rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
}

.emotion-5 {
  margin-bottom: 8px;
}

.emotion-6 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  text-transform: capitalize;
}

.emotion-9 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  display: inline;
  color: #5E696F;
  margin-right: 8px;
}

.emotion-10 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.6;
  letter-spacing: 0.016rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  display: inline;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-weight: 600;
  font-size: 0.9375rem;
  line-height: 1.714;
  text-transform: none;
  letter-spacing: 0.025rem;
  font-family: Open Sans,Segoe UI,Tahoma,sans-serif;
  min-width: 64px;
  padding: 7px 21px;
  border-radius: 6px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(0, 116, 184, 0.5);
  color: #0074B8;
  box-shadow: none;
  width: 100%;
  margin-top: 8px;
}

.emotion-14::-moz-focus-inner {
  border-style: none;
}

.emotion-14.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-14 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 116, 184, 0.04);
  border: 1px solid #0074B8;
}

@media (hover: none) {
  .emotion-14:hover {
    background-color: transparent;
  }
}

.emotion-14.Mui-disabled {
  color: rgba(17, 27, 34, 0.26);
  border: 1px solid rgba(17, 27, 34, 0.12);
}

.emotion-14:hover {
  box-shadow: none;
}

.emotion-14.Mui-focusVisible {
  box-shadow: none;
}

.emotion-14:active {
  box-shadow: none;
}

.emotion-14.Mui-disabled {
  box-shadow: none;
}

.emotion-15 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root emotion-1"
      data-testid="test-clinical-provider-card-42"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <div
          class="MuiBox-root emotion-3"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 emotion-4"
            data-testid="test-provider-name"
          >
            THEODOR GEISEL
          </h6>
          <div
            class="MuiBox-root emotion-5"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-6"
              data-testid="test-provider-address-line1"
            >
              7301 encelia drive
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-6"
              data-testid="test-provider-city-state"
            >
              la jolla, CA 92037
            </p>
          </div>
          <div
            class="MuiBox-root emotion-3"
            data-testid="test-provider-phone-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-9"
            >
              Phone
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-10"
              data-testid="test-provider-phone"
            >
              (999) 555-1212
            </p>
          </div>
          <div
            class="MuiBox-root emotion-3"
            data-testid="test-provider-fax-container"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-9"
            >
              Fax
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 emotion-10"
              data-testid="test-provider-fax"
            >
              (999) 555-1000
            </p>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-disableElevation emotion-14"
          data-testid="test-provider-select-button"
          tabindex="0"
          type="button"
        >
          Select
          <span
            class="MuiTouchRipple-root emotion-15"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
